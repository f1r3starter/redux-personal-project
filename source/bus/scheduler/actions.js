// Types
import { types } from "./types";

export const schedulerActions = {
    // Sync
    fillNewMessage: (message) => {
        return {
            type:    types.FILL_NEW_MESSAGE,
            payload: message,
        };
    },
    clearMessage: () => {
        return {
            type: types.CLEAR_MESSAGE,
        };
    },
};
