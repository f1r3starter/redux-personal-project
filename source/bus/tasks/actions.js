// Types
import { types } from "./types";

export const tasksActions = {
    // Sync
    fillTasks: (tasks) => {
        return {
            type:    types.FILL_TASKS,
            payload: tasks,
        };
    },
    createTask: (task) => {
        return {
            type:    types.CREATE_TASK,
            payload: task,
        };
    },
    removeTask: (taskId) => {
        return {
            type:    types.REMOVE_TASK,
            payload: taskId,
        };
    },
    completeTask: (taskId) => {
        return {
            type:    types.COMPLETE_TASK,
            payload: taskId,
        };
    },
    uncompleteTask: (taskId) => {
        return {
            type:    types.UNCOMPLETE_TASK,
            payload: taskId,
        };
    },
    favoriteTask: (taskId) => {
        return {
            type:    types.FAVORITE_TASK,
            payload: taskId,
        };
    },
    unfavoriteTask: (taskId) => {
        return {
            type:    types.UNFAVORITE_TASK,
            payload: taskId,
        };
    },

    // Async
    fetchTasksAsync: () => {
        return {
            type: types.FETCH_TASKS_ASYNC,
        };
    },
    createTaskAsync: (task) => {
        return {
            type:    types.CREATE_TASK_ASYNC,
            payload: task,
        };
    },
    removeTaskAsync: (taskId) => {
        return {
            type:    types.REMOVE_TASK_ASYNC,
            payload: taskId,
        };
    },
    completeTaskAsync: (taskId) => {
        return {
            type:    types.COMPLETE_TASK_ASYNC,
            payload: taskId,
        };
    },
    uncompleteTaskAsync: (taskId) => {
        return {
            type:    types.UNCOMPLETE_TASK_ASYNC,
            payload: taskId,
        };
    },
    favoriteTaskAsync: (taskId) => {
        return {
            type:    types.FAVORITE_TASK_ASYNC,
            payload: taskId,
        };
    },
    unfavoriteTaskAsync: (taskId) => {
        return {
            type:    types.UNFAVORITE_TASK_ASYNC,
            payload: taskId,
        };
    },
};
