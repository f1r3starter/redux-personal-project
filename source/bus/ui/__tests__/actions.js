// Actions
import { uiActions } from "../actions";

describe("ui actions:", () => {
    test("startSpinning", () => {
        expect(uiActions.startSpinning()).toMatchSnapshot();
    });

    test("stopSpinning", () => {
        expect(uiActions.stopSpinning()).toMatchSnapshot();
    });

    test("emitError", () => {
        expect(uiActions.emitError(__.error, __.meta)).toMatchSnapshot();
    });
});
