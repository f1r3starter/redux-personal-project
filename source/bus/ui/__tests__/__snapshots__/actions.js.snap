// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: emitError 1`] = `
Object {
  "error": true,
  "meta": "123",
  "payload": [Error: TEST_ERROR_MESSAGE.],
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions: startSpinning 1`] = `
Object {
  "type": "START_SPINNING",
}
`;

exports[`ui actions: stopSpinning 1`] = `
Object {
  "type": "STOP_SPINNING",
}
`;
